### PR TITLE
Performance test for event-bus component

### PR DESCRIPTION
**Pull request by anishj0shi**
w.r.t #5224
- ### Performance test for event-bus component (publish service)

#### Scenario at unlimited requests per sec :- 
   * 8 VUS for 90s
   * 16 VUS for 90s
   * 32 VUS for 90s
 
#### K6 Test Results
Screen Shot 2019-08-14 at 1 35 50 PM

### CODE REVIEW

File: tests/perf/prerequisites/event-bus/setup.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
```
Comment (antoineco) [inline tests/perf/prerequisites/event-bus/setup.sh]: Just curious: is there any reason for substituting the Go client with a Bash script? Over time fragmented testing approaches can cause a great deal of headaches.

File: tests/perf/prerequisites/event-bus/setup.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        local response=$(curl -i $SUBSCRIBER_STATUS_URL 2>/dev/null)
+        if [[ $response == *"200"* ]]; then
+            echo -e "\nSubscriber is ready."
+            return 0
+        fi
+        echo -n "."
+        sleep 2
+    done
+    echo -e "\n\n ERROR: Subscriber pod is not ready to listen to events\n"
+    return 1
```
Comment (nachtmaar) [inline tests/perf/prerequisites/event-bus/setup.sh]: Why not use `exit 1` instead to signal that there is an error

Comment (nachtmaar) [reply tests/perf/prerequisites/event-bus/setup.sh]: Ideally we should get a slack notification if the setup of one particular test is failing. Without an exit code !=0 this is hard to detect.

Comment (nachtmaar) [reply tests/perf/prerequisites/event-bus/setup.sh]: Same for other functions

File: tests/perf/components/event-bus/event-bus-perf-test.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            var a = JSON.parse(res.body)
+            if(a && a.status && (a.status == "published")) {
+                return true
+            } else {
+                return false;
+            }
+        }
+    }) ? eventPublishedSuccessfully.add(1) : eventPublishedFailed.add(1);
+
+    // var subscriptionResultResponse = http.get(eventSubscriberUrl);
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/perf/components/event-bus/event-bus-perf-test.js]: not For now, Removed

File: tests/perf/components/event-bus/event-bus-perf-test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    
+  /**
+   * Bursty traffic,
+   * Measure the worst case of bursty traffic, when bursting results in scaling
+   * Bursts: 0 -> N, and N -> 2N
+   * where, N = {1, 100, 1000}
+   */
+    
+  /**
+   * Numbers have to be brought down as the Framework kills the test
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/perf/components/event-bus/event-bus-perf-test.js]: done

File: tests/perf/components/event-bus/event-bus-perf-test.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    /**
+     * Bursty traffic,
+     * Measure the worst case of bursty traffic, when bursting results in scaling
+     * Bursts: 0 -> N, and N -> 2N
+     * where, N = {1, 100, 1000}
+     */
+
+    /**
+     * Numbers have to be brought down as the Framework kills the test
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/perf/components/event-bus/event-bus-perf-test.js]: done

File: tests/perf/components/event-bus/event-bus-perf-test.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    rps: 0,
+    tags: {
+        component: "event-bus",
+        revision: `${__ENV.REVISION}`
+    },
+
+    /**
+     * Bursty traffic,
+     * Measure the worst case of bursty traffic, when bursting results in scaling
+     * Bursts: 0 -> N, and N -> 2N
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/perf/components/event-bus/event-bus-perf-test.js]: added note

File: tests/perf/components/event-bus/event-bus-perf-test.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    Trend
+} from "k6/metrics";
+
+var eventPublishTotalRequests = new Counter("event_publish_requests"),
+    eventPublishedSuccessfully = new Counter("event_publish_success"),
+    eventPublishedFailed = new Counter("event_published_failed"),
+    eventPublishedResponseTime = new Trend("event_publish_response_time", true)
+
+/*
+    Subscription relevant metrics, TO be enabled once subscriber seems to be thread safe
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/perf/components/event-bus/event-bus-perf-test.js]: no it's a concept in progress

File: tests/perf/prerequisites/event-bus/setup.sh [hunk 8]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+set -e
+set -o pipefail
+
+WORKING_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+export NAMESPACE=event-bus-perf-test
+export CLUSTER_DOMAIN=$(kubectl get gateways.networking.istio.io kyma-gateway \
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/perf/prerequisites/event-bus/setup.sh]: done

File: tests/perf/prerequisites/event-bus/setup.sh (hunk 2 above)
Comment (anishj0shi) [reply tests/perf/prerequisites/event-bus/setup.sh]: done

File: tests/perf/components/event-bus/event-bus-perf-test.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+*/
+// var subscriptionResultVerified = new Counter("verified_subscription_results");
```
Comment (Abd4llA) [inline tests/perf/components/event-bus/event-bus-perf-test.js]: Please remove these few commented lines

Comment (anishj0shi) [reply tests/perf/components/event-bus/event-bus-perf-test.js]: done

**Pull request merged**